File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    
    var arr = new Map();
    const n = elements.length;
    
    for(let i=0; i<n; i++){
        var sum = elements[i];
        arr.set(sum);
        for(let j=i+1; j<i+n; j++){
            sum+=elements[j%n];
            arr.set(sum);
        }
    }
    
    return arr.size;
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        
        HashSet<Integer> set = new HashSet<>();
        int n = elements.length;
        
        for(int i=0; i<n; i++){
            int sum = elements[i];
            set.add(sum);
            for(int j=i+1; j<i+n; j++){
                sum += elements[j%n];
                set.add(sum);
            }        
        }

        return set.size();
    }
}